### PR TITLE
enhance: Added optional image and imageAlt to SEO open graph options

### PR DESCRIPTION
**Pull request by santosh898**
This is a small enhancement to use the image from the frontmatter in the Note's SEO open graph options.

This is my for PR. Dropping my discord handle here  - santosh898#7877

### CODE REVIEW

File: packages/common-all/src/types/foundation.ts [hunk 1]
```diff
@@ -183,6 +183,8 @@ export type NoteSEOProps = {
   updated: number;
   created: number;
   excerpt?: string;
+  image?: string;
```
Comment (kevinslin) [inline packages/common-all/src/types/foundation.ts:186]: can you add some docstrings describing these properties?

File: packages/nextjs-template/components/DendronSEO.tsx [hunk 2]
```diff
@@ -47,6 +47,9 @@ export default function DendronSEO() {
   const seoProps = NoteUtils.getSEOProps(noteActive);
   const title = seoProps.title;
   const description = seoProps.excerpt || config.site.description;
+  const images = seoProps.image
+    ? [{ url: seoProps.image, alt: seoProps.imageAlt }]
```
Comment (kevinslin) [inline packages/nextjs-template/components/DendronSEO.tsx:51]: will there be a problem if `url` is set and `imageAlt` is undefined?

**Review by kevinslin — CHANGES_REQUESTED** [review]
awesome to see your first PR in :)

some minor nitpicks and questions

File: packages/common-all/src/types/foundation.ts (hunk 1 above)
Comment (santosh898) [reply packages/common-all/src/types/foundation.ts:186]: Will do

Comment (santosh898) [reply packages/common-all/src/types/foundation.ts:186]: Will do

File: packages/nextjs-template/components/DendronSEO.tsx (hunk 2 above)
Comment (santosh898) [reply packages/nextjs-template/components/DendronSEO.tsx:51]: Looks like it does. I followed `next-seo` type declarations and did this. But now i dug into the [OpenGraph Sepc](https://ogp.me/#structured) and the alt is required.

Should we move the image from `frontMatter.custom` to `frontMatter.image?.` and change image from `string` to `{ url: string, alt: string}`?

Comment (santosh898) [reply packages/nextjs-template/components/DendronSEO.tsx:51]: Or using the note title as the alt prop's fallback is easier?

Comment (SeriousBug) [reply packages/nextjs-template/components/DendronSEO.tsx:51]: Adding this as an actual property to frontmatter sounds like a better design to me. Here's some docs on how to go about adding it: https://wiki.dendron.so/notes/3c80629b-4048-48f0-bdfd-352645bda2ec.html#adding-a-new-frontmatter-property

File: packages/common-all/src/types/foundation.ts (hunk 1 above)
Comment (santosh898) [reply packages/common-all/src/types/foundation.ts]: Hey @kevinslin i adeed the docString for the new image prop.

File: packages/nextjs-template/components/DendronSEO.tsx (hunk 2 above)
Comment (santosh898) [reply packages/nextjs-template/components/DendronSEO.tsx]: @kevinslin and @SeriousBug please take a look, i've added the `image` as an optional property to the frontmatter.

**Review by kevinslin — APPROVED** [review]

**Review by kevinslin — APPROVED** [review]

**Pull request merged**